src/components/pages/AdminCredential/AdminCredentialElements.tsx

```diff
@@ -56,18 +52,45 @@ enum StatusType {
 
 export default function AdminCredentialElements() {
   const { t } = useTranslation()
+  const dispatch = useDispatch()
 
-  const [group, setGroup] = useState<string>(FilterType.ALL)
-  const [expr, setExpr] = useState<string>('')
   const [refresh, setRefresh] = useState<number>(0)
+  const [group, setGroup] = useState<string>(FilterType.ALL)
+  const [searchExpr, setSearchExpr] = useState<string>('')
+  const [filterStatus, setFilterStatus] = useState<string>()
+  const [filterValueAPI, setFilterValueAPI] = useState<string>('')
+  const [fetchHookArgs, setFetchHookArgs] = useState({})
 
   const [getDocumentById] = useFetchNewDocumentByIdMutation()
   const [approveCredential] = useApproveCredentialMutation()
   const [declineCredential] = useDeclineCredentialMutation()
 
   const setView = (e: React.MouseEvent<HTMLInputElement>) => {
     const viewValue = e.currentTarget.value
+    if (viewValue === FilterType.OPEN)
+      setFilterValueAPI(SubscriptionStatus.PENDING)
+    if (viewValue === FilterType.CONFIRMED)
+      setFilterValueAPI(SubscriptionStatus.ACTIVE)
+    if (viewValue === FilterType.DECLINED)
+      setFilterValueAPI(SubscriptionStatus.INACTIVE)
+    setFilterStatus(viewValue)
     setGroup(viewValue)
+    setRefresh(Date.now())
+  }
+
+  useEffect(() => {
+    if (onValidate(searchExpr)) {
+      setFetchHookArgs({
+        filterType: filterValueAPI,
+        expr: searchExpr,
+      })
+    }
+  }, [filterStatus, searchExpr])
+
+  const onValidate = (expr: string) => {
+    const validateExpr = /^[ A-Za-z0-9]*$/.test(expr)
```

Comment:
never use a `*` or `+` in a regex because that means it can have unlimited length which is almost never what we want. Use meaningful lower and upper limits instead. For example if the text must have at least 1 but not more than 2000 characters: `{1,2000}`